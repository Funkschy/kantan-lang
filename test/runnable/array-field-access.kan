import "io";

type Type struct {
    i: i32
}

type Other struct {
    t: Type
}

def main() {
    let a: [2]Type = undefined;
    a[0] = Type { i: 42 };
    a[1].i = (a[0].i = 4) + 1;
    io.printf("%d\n", a[0].i);
    io.printf("%d\n", a[1].i);
    a[1].i += 1;
    io.printf("%d\n", a[1].i);

    let o: [2]Other = undefined;
    o[0].t = a[0];
    io.printf("%d\n", o[0].t.i);
    o[1].t.i = o[0].t.i = 1;
    io.printf("%d\n", o[0].t.i);
    io.printf("%d\n", o[1].t.i);
}
