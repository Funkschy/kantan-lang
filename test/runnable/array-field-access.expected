4
5
6
4
1
1
